src/main/java/tech/sourced/siva/IndexReader.java

```diff
@@ -0,0 +1,119 @@
+package tech.sourced.siva;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.file.attribute.FileTime;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+public class IndexReader {
+    private static final int INDEX_VERSION = 1;
+    private static final int INDEX_FOOTER_SIZE = 24;
+    private static final byte[] INDEX_SIGNATURE = {'I', 'B', 'A'};
+
+    private final RandomAccessFile sivaFile;
+
+    IndexReader(RandomAccessFile sivaFile) throws IOException {
+        this.sivaFile = sivaFile;
+    }
+
+    /**
+     * getFilteredIndex will return an index with all the deleted or modified elements already filtered.
+     *
+     * @return Index with the last entry of each file. Delete files will not appear in this index.
+     * @throws IOException
+     * @throws SivaException
```

Comment:
Is there a reason not to wrap IOException in SivaException? If so, let's document it here to set client expectations.

Otherwise, do you think it's consistent to have 2 separate exceptions here?

In the case of checked exceptions, it would help if we document the recovery steps a client can make as well.